File: 剑指 Offer 03. 数组中重复的数字.cpp
class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
    	unordered_map<int, int> m;
    	for (int i = 0; i < unms.size(); ++i)
    	{
    		if(++m[nums[i]] > 1) return nums[i];
    	}
    return 0;
    }
};

File: CPython/剑指 Offer 03. 数组中重复的数字.py
class Solution:
    def findRepeatNumber(self, nums):
    	res_dict = {}
    	for item in nums:
    		if item  in res_dict:
    			return item
    		else:
    			res_dict[item]+=1
   
    			

if __name__ == '__main__':
	input = [1,2,3,4,1,2,3]
	solution = Solution()
	result = solution.findRepeatNumber(input)
	print(result)
